src/common/core/librarycore.js

```diff
@@ -216,6 +216,288 @@ define([
                 self.setProperty(overlay, to, fromOverlay);
             }
 
+            function isPathInSubTree(fullPath, subTreePath) {
```

Comment:
Couldn't this simply check:
`fullPath.length >= subTreePath.length && fullPath.indexOf(subTreePath) === 0`